enumerable_methods.rb

```diff
@@ -0,0 +1,237 @@
+      end
+    elsif pattern
+      length.times do |i|
+        return true if i == pattern
+      end
+    end
+    false
+  end
+  # rubocop:enable Metrics/PerceivedComplexity,Metrics/CyclomaticComplexity
+
+  # my_none
+  # rubocop:disable Metrics/PerceivedComplexity,Metrics/CyclomaticComplexity
+  def my_none?(pattern = nil)
+    if !block_given? && pattern.nil?
+      length.times do |i|
+        return false if i
+      end
+    elsif self.class == Array
+      length.times do |i|
+        return false if yield(self[i])
+      end
+    elsif self.class == Hash
+      keys = self.keys
+      keys.length.times do |i|
+        return false if yield(keys[i], self[keys[i]])
+      end
+    elsif pattern.is_a? Regexp
+      length.times do |i|
+        return false if i =~ pattern
+      end
+    elsif pattern.is_a? Class
+      length.times do |i|
+        return false if i.is_a? pattern
+      end
+    elsif pattern
+      length.times do |i|
+        return false if i == pattern
+      end
+    end
+    true
+  end
+  # rubocop:enable Metrics/PerceivedComplexity,Metrics/CyclomaticComplexity
+
+  # my_count
+  # rubocop:disable Metrics/PerceivedComplexity,Metrics/CyclomaticComplexity
+  def my_count(arg = nil)
+    return length unless block_given? || !arg.nil?
+
+    counter = 0
+    if !arg.nil?
+      my_each do |i|
+        counter += 1 if i == arg
+      end
+    elsif self.class == Array
+      my_each do |i|
+        counter += 1 if yield(i)
+      end
+    elsif self.class == Hash
+      keys = self.keys
+      my_each do |i|
+        counter += 1 if yield(keys[i], self[keys[i]])
+      end
+    end
+    counter
+  end
+  # rubocop:enable Metrics/PerceivedComplexity,Metrics/CyclomaticComplexity
+
+  # my_map
+  # rubocop:disable Metrics/PerceivedComplexity,Metrics/CyclomaticComplexity
+  def my_map(param = nil)
+    return to_enum unless block_given?
+
+    i = 0
+    arr = []
+    while i < size
+      if block_given? && param.nil?
+        arr << yield(self[i])
+      elsif block_given? && param
+        arr << param.call(self[i])
+      end
+      i += 1
+    end
+    arr
+  end
+  # rubocop:enable Metrics/PerceivedComplexity,Metrics/CyclomaticComplexity
+
+  # my_inject
+  # rubocop:disable Metrics/PerceivedComplexity,Metrics/CyclomaticComplexity
+  def my_inject(*initial)
+    result = nil
+    if block_given?
+      arr = dup.to_a
+      result = initial[0].nil? ? arr[0] : initial[0]
+      arr.shift if initial[0].nil?
+      arr.length.times do |i|
+        result = yield(result, i)
+      end
+    elsif !block_given?
+      arr = dup.to_a
+      if initial[1].nil?
+        sym = initial[0]
+        result = arr[0]
+        arr[1..-1].my_each do |i|
+          result = result.send(sym, i)
+        end
+      elsif !initial[1].nil?
+        sym = initial[1]
+        result = initial[0]
+        arr.my_each do |i|
+          result = result.send(sym, i)
+        end
+      end
+    end
+    result
+  end
+end
```

Comment:
**[REQUIRED]**
* Combines all elements of enum by applying a binary operation, specified by a block.
```
Failure/Error: expect(array.my_inject(&operation)).to eq array.inject(&operation)
      
        expected: 402
             got: 4853
```